client/my-sites/domains/domain-management/transfer/transfer-page/index.tsx

```diff
@@ -72,15 +80,36 @@ const TransferPage = ( props: TransferPageProps ) => {
 		isDomainInfoLoading,
 		isDomainLocked,
 		isDomainOnly,
+		isLoadingPurchase,
 		isMapping,
 		isSupportSession,
+		purchase,
 		selectedDomainName,
 		selectedSite,
 	} = props;
 	const { __ } = useI18n();
 	const [ isRequestingTransferCode, setIsRequestingTransferCode ] = useState( false );
 	const [ isLockingOrUnlockingDomain, setIsLockingOrUnlockingDomain ] = useState( false );
+	const [ showUpsellDiscount, setShowUpsellDiscount ] = useState( false );
 	const domain = getSelectedDomain( props );
+	const canUsePreventTransferPromo =
+		domain?.availablePromos?.length && domain?.availablePromos?.length > 0;
```

Comment:
```suggestion
	const canUsePreventTransferPromo = domain?.availablePromos?.length > 0;
```
Due to the optional chaining operator, I think we can do this directly (`undefined > 0` returns `false`)